### DISCUSSION TITLE
Incoming Message Return Path

### DISCUSSION THREAD

**Discussion by Ramon-LibertyPumps** [postalserver/postal#1954] [Help with using Postal]
I noticed that incoming messages are having the return path re-written to something@psrp.mydomain.com. Is this intended behavior?

The only reason I could think to do this is to eliminate backscatter, but I wonder what other consequences it could have.

Thanks.

**Comment by uhlhosting** [comment]
Yes that is normal.

**Comment by willpower232** [comment]
Yes Postal should be aware of automated replies to the message so it can inform the original system of what happened to the message but it likely doesn't want to receive all emails for a given domain so having a separate subdomain allows everything to receive the email it should.

**Comment by Trapulo** [comment]
So it is mandatory to expose Postal port 25 to Internet also if I need to send email only starting from internal hosts?

**Reply by willpower232** [reply]
if you're emailing anyone on the internet, then they need to be able to access the return path.

**Reply by Trapulo** [reply]
I see, thanks. I think this will be safe because Postal will discard any message that is not an NDR or response to sent emails. Is this right?

**Comment by Ramon-LibertyPumps** [comment]
While I understand the intent of this behavior, I still wish I could turn this feature off for incoming messages.

Two products that we use in between Postal and our Exchange server - Spam Titan and MetaDefender Email Gateway Security - both display quarantine/history using the Envelope-From instead of the message From. This behavior of Postal makes it impossible to search any incoming messages by sender.

**Reply by willpower232** [reply]
If a message has been received from psrp then its a bounce message that needs different handling. Postal should be able to handle normal incoming email sent directly to it.

**Reply by Ramon-LibertyPumps** [reply]
It does handle normal incoming mail, but when it forwards it to the next server in the chain, it re-writes the envelope-from to point to itself rather than leave the original envelope-from that the sender's server used.
